### PR TITLE
[Try] Separate WP API Integration ( Part-1 ) 

### PR DESCRIPTION
**Pull request by sayedtaqui**
## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: packages/story-editor/src/app/api/apiProvider.js [hunk 1]
```diff
@@ -69,6 +43,28 @@ function APIProvider({ children }) {
...
+      autoSaveById,
+      getMedia,
+      uploadMedia,
+      updateMedia,
+      deleteMedia,
+      getLinkMetadata,
+      getAuthors,
+      getCurrentUser,
+      updateCurrentUser,
+      saveMetaBoxes,
```
Comment (swissspidy) [inline packages/story-editor/src/app/api/apiProvider.js:62]: General note / reminder that we should move all meta box related functionality to `wp-story-editor`, including `useSaveMetaBoxes`. This API provider should not know about meta boxes.

File: packages/story-editor/src/editorApp.js [hunk 2]
```diff
@@ -52,23 +50,21 @@ import { GlobalStyle as CalendarStyle } from './components/form/dateTime/calenda
 import KeyboardOnlyOutlines from './utils/keyboardOnlyOutline';
 import { MetaBoxesProvider } from './integrations/wordpress/metaBoxes';
```
Comment (swissspidy) [inline packages/story-editor/src/editorApp.js:51]: Another meta boxes related component that should move to `wp-story-editor`. Just mentioning here so we don't forget it.

File: packages/story-editor/src/index.js [hunk 3]
```diff
@@ -33,4 +36,5 @@ export { default as theme } from './theme'; // @todo To be refactored.
 export { GlobalStyle as CropMoveableGlobalStyle } from './components/moveable/cropStyle';
 export { ConfigProvider as EditorConfigProvider } from './app/config';
```
Comment (swissspidy) [inline packages/story-editor/src/index.js:37]: Isn't that covered by `export * from './app/config';` above btw?

File: packages/story-editor/src/editorApp.js [hunk 4]
```diff
@@ -83,6 +79,7 @@ function App({ config }) {
                                       <CalendarStyle />
                                       <KeyboardOnlyOutlines />
                                       <Layout />
+                                      {children}
```
Comment (swissspidy) [inline packages/story-editor/src/editorApp.js:82]: 👍 

This unlocks a lot!

File: packages/story-editor/src/index.js [hunk 5]
```diff
@@ -17,10 +17,13 @@
 /**
  * Internal dependencies
  */
-import App from './editorApp';
+import { default as StoryEditor } from './editorApp';
+import PostLock from './components/postLock';
+import StatusCheck from './components/statusCheck';
```
Comment (swissspidy) [inline packages/story-editor/src/index.js:22]: Just noting that we might consider just moving these to `wp-story-editor` in the future as these could be considered WP-specific.

File: packages/wp-story-editor/src/index.js [hunk 6]
```diff
@@ -35,38 +38,42 @@ import { initializeTracking } from '@web-stories-wp/tracking';
...
     </FlagsProvider>,
     appElement
   );
 };
 
 const initializeWithConfig = () => {
-  const { id, config, flags } = window.webStoriesEditorSettings;
-  initialize(id, config, flags);
+  const { id, config: settings, flags } = window.webStoriesEditorSettings;
+  initialize(id, settings, flags);
```
Comment (swissspidy) [inline packages/wp-story-editor/src/index.js:76]: Dealing with both `config` and `settings` is quite confusing, especially when doing things like `config: settings` or `<Context.Provider value={{ ...settings, config }}>`, so everything is merged anyway.

Can we just use either `settings` or `config` across the board, not both?

File: packages/wp-story-editor/src/index.js [hunk 7]
```diff
@@ -35,38 +38,42 @@ import { initializeTracking } from '@web-stories-wp/tracking';
 /**
  * Internal dependencies
  */
+import * as apiConfig from './api';
```
Comment (swissspidy) [inline packages/wp-story-editor/src/index.js:41]: Overall I like splitting up the API functions like that 👍 

It's not as low-level as the `dataAdapter` approach in `packages/dashboard/src/app/api/apiProvider.js`.

So once this PR makes it in, we should consider refactoring the dashboard in a similar way.

File: packages/story-editor/src/app/api/apiProvider.js [hunk 8]
```diff
@@ -442,88 +125,116 @@ function APIProvider({ children }) {
...
-      getCustomPageTemplates,
-      deletePageTemplate,
-      getPageTemplates,
-      getCurrentUser,
-      updateCurrentUser,
-    },
-  };
-
-  return <Context.Provider value={state}>{children}</Context.Provider>;
+  // @todo Handle undefined actions in their components.
```
Comment (swissspidy) [inline packages/story-editor/src/app/api/apiProvider.js:128]: Can you elaborate on that?

File: packages/wp-story-editor/src/index.js (hunk 6 above)
Comment (sayedtaqui) [reply packages/wp-story-editor/src/index.js:76]: Yes, I agree, this was just an attempt to separate callbacks and settings from `webStoriesEditorSettings`. Probably I should just go with `config`.

File: packages/story-editor/src/app/api/apiProvider.js (hunk 8 above)
Comment (sayedtaqui) [reply packages/story-editor/src/app/api/apiProvider.js:128]: When some callbacks are not provided, let's say by the playground integration, it should not throw errors and I would add checks to handle these `undefined` callbacks in the components they have been used in.

**Pull request closed without merging**